presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/NullableSliceStateSerializer.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import io.airlift.slice.DynamicSliceOutput;
+import io.airlift.slice.Slice;
+import io.airlift.slice.SliceInput;
+import io.airlift.slice.SliceOutput;
+
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+
+public class NullableSliceStateSerializer
+        implements AccumulatorStateSerializer<NullableSliceState>
+{
+    @Override
+    public Type getSerializedType()
+    {
+        return VARBINARY;
+    }
+
+    @Override
+    public void serialize(NullableSliceState state, BlockBuilder out)
+    {
+        if (state.isNull()) {
+            out.appendNull();
+        }
+        else {
+            SliceOutput sliceOutput = new DynamicSliceOutput((int) state.getEstimatedSize());
+            sliceOutput.writeInt(state.getSlice().length());
+            sliceOutput.writeBytes(state.getSlice());
+            Slice slice = sliceOutput.slice();
+            VARBINARY.writeSlice(out, slice);
```

Comment:
See my comment about including the `type` in the serializer. Then you can use it here